src/java/org/apache/cassandra/cql3/constraints/ScalarColumnConstraint.java

```diff
@@ -125,11 +127,18 @@ protected void internalEvaluate(AbstractType<?> valueType, ByteBuffer columnValu
     @Override
     public void validate(ColumnMetadata columnMetadata) throws InvalidConstraintDefinitionException
     {
+        returnType = columnMetadata.type;
+
+        if (!columnMetadata.name.equals(columnName))
```

Comment:
I think this is a dead code

````
cassandra@cqlsh> CREATE TABLE ks.tb6 (id int primary key, val int check val2 > 5) ;
cassandra@cqlsh> DESCRIBE ks.tb6;

CREATE TABLE ks.tb6 (
    id int PRIMARY KEY,
    val int CHECK val > 5 -- here it will set it to `val` anyway
) 
````

Because we are binding it later on, we do not need to check it here.

But at the same time, it would be appropriate to extract this logic closer to CQL layer (in Cassandra Java code) because we want to still fail to tell user it is invalid.

Reply:
Actually, it has to stay. We still need this for exactly I wrote above.